src/main/java/com/example/springbootbookstore/service/BookServiceImpl.java

```diff
@@ -24,7 +30,7 @@ public BookDto save(CreateBookRequestDto requestDto) {
 
     @Override
     public List<BookDto> findAll() {
-        return bookMapper.toBookDto(bookRepository.findAll());
+        return bookMapper.toBookDto(bookRepository.findAllByIsDeletedFalse());
```

Comment:
```suggestion
        return bookMapper.toBookDto(bookRepository.findAll());
```
default jpa method will return all entites where deleted is false due to the SQLRestriction

Reply:
fixed